File: src/main/java/work/xeltica/craft/core/utils/LocationComparator.java
package work.xeltica.craft.core.utils;

import org.bukkit.Location;

/**
 * 2つのLocationが一致しているかどうかを計算します。
 * @author Xeltica
 */
public class LocationComparator {
    public static boolean equals(Location l1, Location l2) {
        if (!l1.getWorld().equals(l2.getWorld())) return false;

        final var x1 = l1.getX();
        final var y1 = l1.getY();
        final var z1 = l1.getZ();

        final var x2 = l2.getX();
        final var y2 = l2.getY();
        final var z2 = l2.getZ();
        return x1 == x2 && y1 == y2 && z1 == z2;
    }
}

File: src/main/java/work/xeltica/craft/core/api/LocationComparator.kt
package work.xeltica.craft.core.api

import org.bukkit.Location

/**
 * 2つのLocationが一致しているかどうかを計算します。
 * @author Xeltica
 */
object LocationComparator {
    fun equals(l1: Location, l2: Location): Boolean {
        if (l1.world != l2.world) return false
        val x1 = l1.x
        val y1 = l1.y
        val z1 = l1.z
        val x2 = l2.x
        val y2 = l2.y
        val z2 = l2.z
        return x1 == x2 && y1 == y2 && z1 == z2
    }
}